pkg/shared/client.go

```diff
@@ -16,19 +16,25 @@ package shared
 
 import (
 	"github.com/sirupsen/logrus"
+	"k8s.io/api/core/v1"
 	"k8s.io/client-go/kubernetes"
+	"k8s.io/client-go/kubernetes/scheme"
+	typedcorev1 "k8s.io/client-go/kubernetes/typed/core/v1"
 	"k8s.io/client-go/rest"
+	"k8s.io/client-go/tools/record"
 )
 
 var (
-	Client *K8sClient
+	Client        *K8sClient
+	eventRecorder record.EventRecorder
 )
 
 // init creates the REST API client that will be used for actions not
 // supported through the operator SDK.
```

Comment:
This header comment can also mention that it creates the event recorder.

BTW I know we talked a bit offline about whether the event recorder handle should instead be stored in the Handler structure. I think it's fine/good to have it here since it's an infrastructural thing only needed by this module's internal code, as opposed to the stuff in Handler which gets passed around everywhere.